erpnext/manufacturing/doctype/production_plan/test_production_plan.py

```diff
@@ -679,7 +679,14 @@ def test_qty_based_status(self):
 		self.assertFalse(pp.all_items_completed())
 
 	def test_production_plan_planned_qty(self):
-		pln = create_production_plan(item_code="_Test FG Item", planned_qty=0.55)
+		from erpnext.stock.doctype.item.test_item import make_item
+
+		fg_item = make_item(properties={"is_stock_item": 1, "stock_uom": "_Test UOM 1"}).name
+		bom_item = make_item().name
+
+		make_bom(item=fg_item, raw_materials=[bom_item], source_warehouse="_Test Warehouse - _TC")
+
+		pln = create_production_plan(item_code=fg_item, planned_qty=0.55)
```

Comment:
THe issue is not about making this test pass or using correct values here. The issue actually applies to end user too. 

If you use Nos and set qty as 0.5 production plan currently allows doing that... which doesn't make sense. 

<img width="1046" alt="Screenshot 2022-06-22 at 12 25 28 PM" src="https://user-images.githubusercontent.com/9079960/174963660-9fa0d77a-a042-4cb7-ac1a-846827532639.png">